kvdb/postgres/db.go

```diff
@@ -57,11 +57,21 @@ type db struct {
 // Enforce db implements the walletdb.DB interface.
 var _ walletdb.DB = (*db)(nil)
 
-// Global set of database connections.
-var dbConns *dbConnSet
+var (
+	// dbConns is a global set of database connections.
+	dbConns   *dbConnSet
+	dbConnsMu sync.Mutex
```

Comment:
Is this commit fixing an existing bug, or is it a preparation for what's coming next? Would be nice to have that info in the commit message.

Reply:
The file is `kvdb/postgres`, so it was clear all along that it was pre-existing, sorry.

Reply:
Only now that I look better at the master code, I do wonder what the actual race condition was. It seems that `dbConns` was only set in `postgres.Init` and that that function is only called once?

Reply:
I think it's just a precaution, noticed during the initial review of this PR: https://github.com/lightningnetwork/lnd/pull/7251#discussion_r1048889003

Reply:
I see. Yes, I remember that I did consider race conditions when writing this code and thought it was safe back then. But a precaution doesn't hurt.